narwhals/_compliant/dataframe.py

```diff
@@ -99,7 +115,13 @@ def from_numpy(
         schema: Mapping[str, DType] | Schema | Sequence[str] | None,
     ) -> Self: ...
     def __array__(self, dtype: Any, *, copy: bool | None) -> _2DArray: ...
-    def __getitem__(self, item: Any) -> CompliantSeriesT | Self: ...
+    def __getitem__(
+        self,
+        item: tuple[
+            SingleIndexSelector | MultiIndexSelector[CompliantSeriesT],
+            MultiIndexSelector[CompliantSeriesT] | MultiColSelector[CompliantSeriesT],
```

Comment:
```suggestion
            SingleColSelector | MultiColSelector[CompliantSeriesT],
```
Not sure. But it sounds more logical to me that way (note, I'm a 100% narwhals newbie).

Reply:
Thanks for your suggestion!

I don't think it's right though, the singlecolselector case returns a Series and that's handled separately